content/blog/2024-06-12-rustl-2024-unconference.md

```diff
@@ -0,0 +1,200 @@
+
+- Build tooling (especially for Android and other non-desktop platforms).
+- Text layout and editing.
+- Winit adoption.
+- Accessibility.
+- Potential for further collaboration.
+- Requests for new Rust features.
+
+Note that I'm skipping a lot topics on the agenda, and other side discussions.
+It feels like those topics were the most prominent during those two days.
+
+There was also a fair amount of informal chatter around LogLog Games' strongly critical [Leaving Rust gamedev after 3 years](https://loglog.games/blog/leaving-rust-gamedev/) article.
+I think those discussions mostly reflected the Bevy community's consensus, that the article had some flaws but overall pointed at very clear weaknesses in the Rust GUI / game engine ecosystem.
+
+### Build tooling
+
+Everyone agreed that a better solution was needed to ship non-TUI Rust apps.
+
+There was discussion about various platforms and build system, and the difficulty of having to eg write Java code for an Android port, and whether anyone wanted to take responsibility for maintaining that glue layer for the rest of the ecosystem.
+
+No actionable decisions were made, but consensus seems to be that although integrated build tools are desirable, external build tooling for complex platforms (primarily Apple and Android) is likely to remain necessary.
+Nobody seems to have stepped up to work on multi-target builds in Cargo, and since external build tools are required as it is, most people are okay with building everything several times when they release a multi-target package.
+Most users are comfortable invoking platform-specific tools for bundling, signing, and manifest generation, so there is not much urgency for integrated tooling for these.
+
+### Text layout and editing
+
+There was some widespread interest in the Linebender ecosystem's text handling crates.
+From what I remember, Rik Arends and others had concerns about whether eg Parley and Swash depend on our Vello renderer (they don't).
+
+Some people also wanted Parley to be decoupled from Swash.
+
+People we especially interested in the idea of having access to a common text-editing widget, or at least infrastructure to create one.
+That widget would handle IME, text selection, accessibility actions, etc, in a way that would feel native to different platforms with their own text-editing quirks (mobile, MacOS, Linux with vim mode, etc).
+
+### Winit
+
+By now all of the Rust ecosystem has firmly converged on [`winit`](https://github.com/rust-windowing/winit) as the windowing platform-abstraction solution of choice.
+
+(Well, not all! [One small project](https://github.com/makepad/makepad) with indomitable maintainers still holds out against the invaders.)
+
+The consensus during this discussion seemed to be that Winit was there to stay.
+[TAO](https://github.com/tauri-apps/tao) users present were generally interested in being able to use upstream winit in the future, if its API adopts or supersedes the improvements made in TAO.
```

Comment:
```suggestion
[TAO](https://github.com/tauri-apps/tao) users present were generally interested in being able to use upstream `winit` in the future, if its API adopts or supersedes the improvements made in TAO.
```

Reply:
Yeah, that or - at least, a scan of https://docs.rs/winit/latest/winit/ suggests it's now a proper noun; see "All of Winit’s functions"
```suggestion
[TAO](https://github.com/tauri-apps/tao) users present were generally interested in being able to use upstream Winit in the future, if its API adopts or supersedes the improvements made in TAO.
```